src/components/flat-table/flat-table.pw.tsx

```diff
@@ -0,0 +1,3619 @@
+    mount,
+    page,
+  }) => {
+    await mount(<FlatTableAllSubrowSelectableComponent />);
+
+    await actionPopoverButton(page).nth(0).click();
+    await expect(actionPopover(page)).toHaveCount(1);
+  });
+
+  test(`should render with action popover in a cell opened by Spaceber`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<FlatTableAllSubrowSelectableComponent />);
+
+    const popover = actionPopoverButton(page).nth(0);
+    await popover.focus();
+    await popover.press("Space");
+    await expect(popover).toHaveCount(1);
+  });
+
+  test(`should render with action popover in a cell opened by Enter key`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<FlatTableAllSubrowSelectableComponent />);
+
+    const popover = actionPopoverButton(page).nth(0);
+    await popover.focus();
+    await popover.press("Enter");
+    await expect(popover).toHaveCount(1);
+  });
+
+  ([
+    ["UK", 0, 1],
+    ["UK", 0, 2],
+    ["UK", 0, 3],
+    ["Germany", 1, 2],
+    ["Germany", 1, 3],
+    ["Germany", 1, 0],
+    ["China", 2, 3],
+    ["China", 2, 0],
+    ["China", 2, 1],
+    ["US", 3, 0],
+    ["US", 3, 1],
+    ["US", 3, 2],
+  ] as [string, number, number][]).forEach(
+    ([record, position, destinationId]) => {
+      test(`should drag FlatTable draggable row ${record} and re-order to position ${destinationId}`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(<FlatTableDraggableComponent />);
+
+        const draggableItem = flatTableDraggableItem(page, position);
+        const dropPosition = flatTableDraggableItemByPosition(
+          page,
+          destinationId
+        );
+        await draggableItem.dragTo(dropPosition);
+        await expect(
+          flatTableDraggableItemByPosition(page, destinationId)
+        ).toHaveText(record);
+      });
+    }
+  );
+
+  ([
+    ["large", 700, 345],
+    ["small", 700, 240],
+  ] as [string, number, number][]).forEach(
+    ([size, tableWidth, tableHeight]) => {
+      test(`should render with pager and sticky header in ${size} viewport`, async ({
+        mount,
+        page,
+      }) => {
+        await page.setViewportSize({ width: tableWidth, height: tableHeight });
+        await mount(<FlatTablePagerStickyHeaderComponent />);
+
+        await flatTablePageSizeSelect(page).click();
+
+        if (size === "large") {
+          const selectListPos = flatTablePageSelectListPosition(page);
+          await expect(selectListPos).toHaveAttribute(
+            "data-floating-placement",
+            "bottom"
+          );
+          await expect(selectListPos).toBeInViewport();
+        } else {
+          const selectListPos = flatTablePageSelectListPosition(page);
+          await expect(selectListPos).toHaveAttribute(
+            "data-floating-placement",
+            "top"
+          );
+          await expect(selectListPos).toBeInViewport();
+        }
+      });
+    }
+  );
+
+  ([
```

Comment:
suggestion: same comment about storing these in a variable but again not a huge issue as they're next to each other here etc